src/UnitOfWork.php

```diff
@@ -2366,11 +2365,7 @@ public function createEntity(string $className, array $data, array &$hints = [])
             }
 
             if ($this->isUninitializedObject($entity)) {
-                if ($this->em->getConfiguration()->isNativeLazyObjectsEnabled()) {
-                    $class->reflClass->markLazyObjectAsInitialized($entity);
-                } else {
-                    $entity->__setInitialized(true);
-                }
+                $class->reflClass->markLazyObjectAsInitialized($entity);
 
                 Hydrator::hydrate($entity, (array) $class->reflClass->newInstanceWithoutConstructor());
```

Comment:
We still depend on `Symfony/VarExporter` because of this line :thinking:

Reply:
this line is not needed when using native lazy objects. Native lazy objects already apply default values of properties automatically during initialization.

This line should be in the `else` branch actually (this is probably something missed during the merge-up between 2.20.3 and 3.4.x)

Reply:
I think that was my mistake. Thanks for fixing it @stof , released as https://github.com/doctrine/orm/releases/tag/3.4.3